my-app/src/DataStore.jsx

```diff
@@ -437,6 +448,40 @@ class DataStore {
     return Promise.resolve(Array.from(this._persons).filter(obj => obj[1].name === name).map(obj=>obj[1]));
   }
 
+  getPersonByEmail(email, callback){
+    /**
+     * Gets the person with an email
+     * @param email {string} email to search by
+     * @return {Promise} promise for a {Person}
```

Comment:
Does this necessarily return a promise?

Reply:
`getPersonsByEmail` works differently than `getPersonsByName` or `getPersonsById` - realizing that they should all work the same way. What I'd propose is:
- All return a promise
- All check local memory for the person and then fall back to checking the DB